### ISSUE TITLE
pre-GCN AMD Cards reported hashrate wrong

### ISSUE THREAD

**Issue by dormancygrace** [fireice-uk/xmr-stak#1615]
i`m using old Intel xeon e5430, AMD Radeon HD 5770 1Gb. I achieved best performance with this config: 
"gpu_threads_conf" : [
  // gpu: Juniper memory:384
  // compute units: 10
  { "index" : 0,
    "intensity" : 256, "worksize" : 16,
    "affine_to_cpu" : false, "strided_index" : 2, "mem_chunk" : 2,
    "comp_mode" : true
  },
With this one GPU gives ~135 H/s. I`ve tried to change mem_chunk to 16 and got incredible ~400 H/s. But miner only reports high hashrate, on the pool i can see real hashrate and it`s ~60 (only CPU).

**Comment by Spudz76** [comment]
Yes, drivers newer than 14.4 do not work nice on Terascale and even some GCN1 - also note it is only detecting 384MB of the 1024MB it should have.

I use fglrx 15.x driver in Ubuntu 14 for mine and it shows 896MB which is still missing 128MB somehow, and if given 'bad' intensities it will report strange hashrates outside of reality.  I should revert to 14.x driver but currently don't have that card in the rig and running some Tahiti LE in it on the 15.

AMD broke the Terascale in the driver after 14.x,  and same applies to Windows.

**Comment by Spudz76** [comment]
OK, reverted my Ubuntu 14 rig to `3.19.0-80-generic` kernel and installed `fglrx-14.501.1003` with patch from [here](https://gist.github.com/aginor/4f232d4c28aef7685282)

Same rates from the Tahitis, good.  Redwood still says 896MB so I guess that's normal. Tried several settings but this gives 100H/s for it which seems pretty good (never got more than maybe 75 with any other combos)
```
  // gpu: Redwood memory:896
  // compute units: 5
  { "index" : 0,
    "intensity" : 225, "worksize" : 15,
    "affine_to_cpu" : false, "strided_index" : 1, "mem_chunk" : 2,
    "comp_mode" : true
  },
```
Note the possible harmonic for tuning I used, three times compute units.  So on yours reporting 10 units try 10, 20, 30 worksizes with intensity 240 or around there.  It will underutilize memory still, but I don't think there is enough horsepower in these to run a full memory load of blocks.

**Comment by dormancygrace** [comment]
I`ve already installed 14.4. Issue still exists

**Comment by Spudz76** [comment]
Yeah, don't use strided_index:2, I also had iffy luck with 0.  Terascale do not enjoy for some reason.

Does it work with the settings I suggested?
Does it show better memory amount?
Did you redetect (delete/move amd.txt from any previous runs)?

**Comment by dormancygrace** [comment]
It works best with my settings/ Yes, it shows more memory. but it`s not the case =)

**Comment by Spudz76** [comment]
Learned new, check #472

**Comment by dormancygrace** [comment]
It is different.

**Comment by dormancygrace** [comment]
![image](https://user-images.githubusercontent.com/20384748/41361372-92d0569c-6f37-11e8-93a9-35500081ccb4.png)

**Comment by dormancygrace** [comment]
AMD Radeon HD 5770, 1Gb GDDR5
